### ISSUE TITLE
Outline simple Bedmaster QC pipeline

### ISSUE THREAD

**Issue by erikr** [aguirre-lab/ml4c3#1268]
## What and why
So we can easily check for egregious errors in tensorized Bedmaster data.

## Solution(s)
Outline what data aspects to assess, and the big picture layout of the code.

## Acceptance criteria
List of precisely defined checks, to facilitate further discussion from @aguirre-lab/ml4c3-dev, @matt-cdac, @contactadityag, and our PIs.

**Comment by epalto** [comment]
Right now the BedmasterReader performs QC on the data and filters out signals with wrong format:
* Signals encoded as unicode strings are converted to float. Signals that contained non-sense data are removed, and values like "X--" or "XXX" are converted to nans.
* Vital signs whose time vector doesn't have the same length than the values vector are filtered out
* Waveforms that have no name are filtered out
* Waveforms with multiple names are filtered out (a machine is connected to a channel, then removed and then another machine is connected to the same channel)
* Incorrect sample frequencies are set to Nan
* Waveforms are checked to contain a timestamp associated with each values (with help of samples_per_ts)
* Interbundle correction removes redundant parts of the signals

**Comment by erikr** [comment]
Eric, thanks for the informative comment! It makes me rethink the scope of this issue.

We can categorize the above QC checks into two categories:
1. During tensorization (followed by corrections to the data problems if possible).
2. After tensorization.

We started with nothing in either category and understood nothing about the data.

Over time, we tensorized data and performed checks in category 2. Once we understood the problem and how to fix it, we moved the check into category 1 so the problematic data never arrived in the `hd5`.

Eventually we developed a comprehensive set of QC checks and corrections in our pipeline, which you nicely outlined above. 

Is this correct?

> I am writting some code that for the tensorized files shows:
> - Non-monotonicities: number, duration and length of the time jump

Some non-monotocities will ideally be solved by https://github.com/aguirre-lab/ml4c3/issues/1265. But even after that PR merges and we re-tensorize some data, what should we do if non-monotocities are still detected?

> - Maximum time period without data inside a signal

A signal may contains data with long segments of emptiness. This is acceptable and does not prove the signal was incorrectly tensorized. I propose this is a useful `TensorMap` to develop for `explore` in order to assess data, but should not be a Category 1 QC check and correction.

Aside from #1265, I think we should no longer put cycles into `tensorization` code unless we discover a problem that affects @epalto @raimonpv @RogerPallares thesis work.

In the next few weeks, I propose we deprecate Bedmaster tensorization functionality in `ml4c3`. CDAC will be responsible for tensorizing and providing Bedmaster data, and maintaining their fork of the repo (which should credit our team for development work).

We need to decide when to change our role to **data users**. When that occurs, and we identify problems such as https://github.com/aguirre-lab/ml4c3/issues/1258, we'll notify @contactadityag @matt-cdac instead of fixing it ourselves.

To summarize:
- QC steps are performed by `tensorization` and corrections are made when possible.
- We can update the pipeline to handle non-monotonicities.
- In the next few weeks, we will hand off the `tensorization` pipeline and all responsibility for generation of Bedmaster data to CDAC.
- We should harmonize high-level analysis of Bedmaster data (distinct from pre-tensorized QC). In other words, we need to avoid writing Jupyter Notebooks from scratch, and instead leverage existing code by using `explore` and `TensorMaps`. `explore` was created for this very purpose, and if it is not popular, we must improve it.
- I will close this issue. We can create new and precise issues for updates to `tensorization`.
